### ISSUE TITLE
Filter should be consistent 

### ISSUE THREAD

**Issue by gotenxds** [selfrefactor/rambda#325]
Currently the `filter` method returns an empty array if the second parameter is `undefined`
Not so when sending in `null`

This should be consistent
lodash for example always returns array if you call it with undefined or null I believe we should do the same.

Currently :
```js
filter(() => true, undefined); // []
filter(() => true, null); // {}
```

**Comment by selfrefactor** [comment]
Actually, I'd prefer to throw here, but I also remember that this handling of `undefined` was suggested before by other contributor so I will just add your suggestion. `Ramda` throws in both examplese as you can see here https://ramdajs.com/repl/#?R.filter%28%28%29%20%3D%3E%20true%2C%20null%29%0A

The change will be applied with upcoming version `4.4.0`

**Comment by gotenxds** [comment]
I suggest accepting undefine and null, it makes working with them much more clean instead of constantly checking for it. This also the way lodash is doing it and people coming from it will find this annoying

**Comment by selfrefactor** [comment]
I already applied your suggestion, just waiting for one more task to complete and will release `4.4.0`. Thank you for taking the time to file this issue.

**Comment by selfrefactor** [comment]
Published with `4.4.0`
